contract/models/contract.py

```diff
@@ -43,6 +44,12 @@ class ContractContract(models.Model):
         inverse_name="contract_id",
         copy=True,
     )
+    # Trick for being able to have 2 different views for the same o2m
+    contract_line_fixed_ids = fields.One2many(
+        string="Contract lines (fixed)",
+        comodel_name="contract.line",
+        inverse_name="contract_id",
+    )
```

Comment:
Shouldn't this have a `domain`?

Reply:
No, as both views must contain the same number of lines. It's only for providing 2 different views of the same records (one with all the fields, and the other with simplified fields), which is not possible in regular one2many widget (that can't be put 2 times with different views). The main question here that forbids me to put `attrs` in fields for controlling visibility is that one of the tree must be editable inline, and the other not, so that can't be done through `attrs`. I'm expanding the comment for noting this.